processor/interface.go

```diff
@@ -2,12 +2,22 @@ package processor
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/cartridge-gg/starknet-indexer/ent"
 	"github.com/dontpanicdao/caigo/jsonrpc"
 	"github.com/dontpanicdao/caigo/types"
 )
 
+type ProcessorError struct {
```

Comment:
Can be nice to have a `NewProcessorError(scope string, err error) *ProcessorError` method for constructing the error